File: libgomp/testsuite/libgomp.oacc-c-c++-common/combined-reduction.c
/* Test a combined acc parallel loop reduction.  */

/* { dg-do run } */

#include <assert.h>

int
main ()
{
  int i, v1 = 0, v2 = 0, n = 100;

#pragma acc parallel loop reduction(+:v1, v2)
  for (i = 0; i < n; i++)
    {
      v1++;
      v2++;
    }

  assert (v1 == n);
  assert (v2 == n);

  return 0;
}

File: libgomp/testsuite/libgomp.oacc-fortran/combined-reduction.f90
! Test a combined acc parallel loop reduction.

! { dg-do run }

program test
  implicit none
  integer i, n, var

  n = 100
  var = 0

  !$acc parallel loop reduction(+:var)
  do i = 1, 100
     var = var + 1
  end do
  !$acc end parallel loop

  if (var .ne. n) call abort
end program test
